hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithFailover.java

```diff
@@ -0,0 +1,46 @@
+package org.apache.hadoop.ozone.om;
+
+import org.apache.hadoop.ozone.OzoneConfigKeys;
+import org.apache.hadoop.ozone.om.ha.OMFailoverProxyProvider;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.apache.hadoop.ozone.MiniOzoneOMHAClusterImpl.NODE_FAILURE_TIMEOUT;
+
+/**
+ * Test Ozone Manager operation in distributed handler scenario with failover.
+ */
+public class TestOzoneManagerHAWithFailover extends TestOzoneManagerHA {
+  /**
+   * 1. Stop one of the OM
+   * 2. make a call to OM, this will make failover attempts to find new node.
+   * a) if LE finishes but leader not ready, it retries to same node
+   * b) if LE not done, it will failover to new node and check
+   * 3. Try failover to same OM explicitly.
+   * Now #3 should wait additional waitBetweenRetries time.
+   * LE: Leader Election.
+   */
+  @Test
+  public void testIncrementalWaitTimeWithSameNodeFailover() throws Exception {
```

Comment:
Why move this test into a new class? Is it because it modifies the mini-cluster in a way we cannot easily fix the cluster later?

Reply:
Yes, we were unable to recover the cluster to a proper stable state after this test - which led to it polluting the other test cases. So decided to move it into its own class so that it uses an isolated cluster for itself.

Reply:
OK - makes sense. I think it would be good to add a note in the class level comment that this class should not have any further tests added, as the cluster cannot be reused. This may help someone in the future if they try to add to this class.

Reply:
Yeah, that makes sense. I'll add this note. Thanks!